### DISCUSSION TITLE
Test against libregress?

### DISCUSSION THREAD

**Discussion by BlobCodes** [hugopl/gi-crystal#23] [Ideas]
I think it may be a good idea to test this library against libregress (previously libeverything).
PyGObject example: https://gitlab.gnome.org/GNOME/pygobject/-/blob/master/tests/test_everything.py

**Comment by hugopl** [comment]
I remember reading about libeverything (maybe on GObject Introspection docs), but as I found absolutely no documentation about it, so was easier to just port the the patch that I already had done for https://github.com/jhass/crystal-gobject and use it in gi-crystal.

Is the code for this library auto generated? I tried to find things like "where test_boolean is defined in C?" and wasn't able to find using Gitlab project search.

**Reply by BlobCodes** [reply]
It was manually written.
Here is the link: https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/main/tests/scanner/regress.c

**Comment by hugopl** [comment]
I found where I saw about this https://gi.readthedocs.io/en/latest/writingbindings/guidelines.html

Probably at the time it was already renamed to regress but the text still nowadays mentioning `libeverything`, what explain why I never found it... I also was expecting it to live in a different repository to be even easier to be used by other bindings.

I think we can start using then for new tests and slowly replacing tests using things in `libtest` that have matches in `libregress`, it's possible that we find some specific bugs that aren't covered in `libregress`, so we will probably live with both libraries.

So, to use libregrex we would need to copy regress.[ch] to the repository and tell the linter to forget about these files, since they use a different formatting.

**Comment by hugopl** [comment]
I created a MR to help future people to not repeat my mistake
https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/342/diffs

**Comment by hugopl** [comment]
I changed my mind about this, IIRC libregress depends on cairo for some stuff it tests (under a `_GI_DISABLE_CAIRO` define), anyway... due to the amount of tests we have nowadays I think the time we could spend to let gi-crystal use libregress would be better spent fixing the existing bugs and implementing the tons of lacking features we need.
